File: ide/src/main/java/org/dau/ui/schematic/fx/controls/FxArrow.java
package org.dau.ui.schematic.fx.controls;

import javafx.scene.shape.ClosePath;
import javafx.scene.shape.LineTo;
import javafx.scene.shape.MoveTo;
import javafx.scene.shape.Path;

public final class FxArrow extends Path {

  public final double length;

  public FxArrow(double length, double phi) {
    this.length = length;
    getElements().addAll(
      new MoveTo(0d, -length * Math.sin(phi)),
      new LineTo(0d, length * Math.sin(phi)),
      new LineTo(length * Math.cos(phi), 0d),
      new ClosePath()
    );
  }
}

File: ide/src/main/kotlin/org/dau/ui/schematic/controls/FxArrow.kt
package org.dau.ui.schematic.controls

import javafx.scene.shape.ClosePath
import javafx.scene.shape.LineTo
import javafx.scene.shape.MoveTo
import javafx.scene.shape.Path

class FxArrow(val length: Double, phi: Double) : Path() {

  init {
    elements.addAll(
      MoveTo(0.0, -length * Math.sin(phi)),
      LineTo(0.0, length * Math.sin(phi)),
      LineTo(length * Math.cos(phi), 0.0),
      ClosePath()
    )
  }
}
